crates/lib/src/stdlib/tags/render_tag.rs

```diff
@@ -0,0 +1,474 @@
+        }?;
+
+        arguments.expect_nothing()?;
+
+        Ok(Box::new(Render {
+            partial,
+            expr: for_or_with,
+        }))
+    }
+
+    fn reflection(&self) -> &dyn TagReflection {
+        self
+    }
+}
+
+#[derive(Debug)]
+struct Render {
+    partial: Expression,
+    expr: ForOrWith,
+}
+
+impl Renderable for Render {
+    fn render_to(&self, writer: &mut dyn Write, runtime: &dyn Runtime) -> Result<()> {
+        let value = self.partial.evaluate(runtime)?;
+        if !value.is_scalar() {
+            return Error::with_msg("Can only `include` strings")
+                .context("partial", format!("{}", value.source()))
+                .into_err();
+        }
+        let name = value.to_kstr().into_owned();
+
+        match &self.expr {
+            ForOrWith::With(vars) => {
+                let mut pass_through = std::collections::HashMap::new();
+                if !vars.is_empty() {
+                    for (id, val) in vars {
+                        let value = val
+                            .try_evaluate(runtime)
+                            .ok_or_else(|| Error::with_msg("failed to evaluate value"))?;
+
+                        pass_through.insert(id.as_ref(), value);
+                    }
+                }
+
+                let scope = SandboxedStackFrame::new(runtime, &pass_through);
+
+                let partial = scope
+                    .partials()
+                    .get(&name)
+                    .trace_with(|| format!("{{% render {} %}}", self.partial).into())?;
+
+                partial
+                    .render_to(writer, &scope)
+                    .trace_with(|| format!("{{% render {} %}}", self.partial).into())
+                    .context_key_with(|| self.partial.to_string().into())
+                    .value_with(|| name.to_string().into())?;
+            }
+            ForOrWith::For(range, var_name) => {
+                let range = range
+                    .evaluate(runtime)
+                    .trace_with(|| format!("{{% render {} %}}", self.partial).into())?;
+                let array = range.evaluate()?;
+
+                if !array.is_empty() {
+                    let len = array.len();
+                    for (i, v) in array.into_iter().enumerate() {
+                        let forloop = ForloopObject::new(i, len);
+                        let mut root = std::collections::HashMap::<
+                            liquid_core::model::KStringRef<'_>,
+                            &dyn ValueView,
+                        >::new();
+                        root.insert("foorloop".into(), &forloop);
+                        root.insert(var_name.as_ref(), &v);
+
+                        let scope = SandboxedStackFrame::new(runtime, &root);
+
+                        let partial = scope
+                            .partials()
+                            .get(&name)
+                            .trace_with(|| format!("{{% render {} %}}", self.partial).into())?;
```

Comment:
from [the docs](https://shopify.github.io/liquid/tags/template/#render)

> Note that you don’t need to write the file’s .liquid extension.

Do we need to handle this here?